umbilical/revocation/checker.go

```diff
@@ -59,9 +64,102 @@ func NewChecker(cfg Config) (*Checker, error) {
 
 	ret.crlFetching = make(map[string]*sync.Cond)
 
+	if err := ret.evictExpired(); err != nil {
+		return nil, err
+	}
+
+	// Automatically check for expired cache entries in the background.
+	ret.evictionWorkerChan = make(chan struct{})
+	ret.wg.Add(1)
+	go func() {
+		defer ret.wg.Done()
+		for {
+			select {
+			case <-ret.evictionWorkerChan:
+				return
+			case <-time.After(time.Hour):
+			}
+
+			if err := ret.evictExpired(); err != nil {
+				slog.Error(
+					"evictExpired",
+					"err",
+					err,
+				)
+			}
+		}
+	}()
+
 	return &ret, nil
 }
 
+// Walks cache and removes expired entries.
+func (c *Checker) evictExpired() error {
+	toRemove := make(map[string]uint64)
+
+	totalEntries := 0
+	removedEntries := 0
+	startTime := time.Now()
+	err := c.cache.View(func(tx *bolt.Tx) error {
+		crlsBucket := tx.Bucket([]byte("crls"))
+		c := crlsBucket.Cursor()
+
+		var entry crlEntry
+		for url, rawEntry := c.First(); url != nil; url, rawEntry = c.Next() {
+			totalEntries++
+			if err := json.Unmarshal(rawEntry, &entry); err != nil {
+				return err
+			}
+			if time.Until(entry.Expires) < 0 {
+				toRemove[string(url)] = entry.BucketID
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("iterating expired CRLs: %w", err)
+	}
+
+	if len(toRemove) == 0 {
+		slog.Info(
```

Comment:
You can also do `slog.Info("msg", slog.Any("a", a), slog.Any("b", b))` to keep the KV pairs grouped together a little more closely (but either way is fine with me).